### ISSUE TITLE
Processing only one language

### ISSUE THREAD

**Issue by CosmoLuminous** [wasiahmad/PLBART#40]
Hi @wasiahmad, 

I have following queries:

1. Could you please guide me on the steps if we want to pre-train the model only for the downstream tasks which require only one Programming language (Eg. Clone detection). Does this library incorporates this scenario, or do I need to make certain changes to accommodate it? 

2. Also, the use of number of GPUs has been hardcoded for 8 GPU and not taken as an input argument from the user. Will this work if we have only 1 or 2 GPUs on the system?

3. What is the early stopping criteria for pre-training? What is the hyperparameter for the same

Thank you,
Aman

**Comment by wasiahmad** [comment]
Hello,

1. Yes, you need to make small changes. For example, you can only use your target language(s) during pretraining as mentioned [here](https://github.com/wasiahmad/PLBART/blob/main/pretrain/pretrain.sh#L8). If you do not need a certain language, just skip preparing its corresponding pretraining data.
2. This is not true. You can always vary the number of GPUs (see [here](https://github.com/wasiahmad/PLBART/blob/main/pretrain/pretrain.sh#L29)). Please, make sure to adjust other hyper-parameters to achieve the desired batch size.
3. We used a fixed number of pre-training steps (100k). You can use validation loss as a stopping criterion too.

Thanks!

**Comment by CosmoLuminous** [comment]
Thank you for your response.

regards,
Aman
